ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiUserManagementService.scala

```diff
@@ -431,12 +431,13 @@ private[apiserver] final class ApiUserManagementService(
           )
       }
 
-  // Check if user either doesn't exist or exists and belongs to the requested Identity Provider
+  // Check if user either doesn't exist or if it does - it belongs to the requested Identity Provider
   // Alternatively, identity_provider_id could be part of the compound unique key within user database.
   // It was considered as complication for the implementation and for now is simplified.
-  private def verifyUserIsNonExistentOrInIdp(
+  private def verifyUserDoesNotExist(
       identityProviderId: IdentityProviderId,
       userId: Ref.UserId,
+      operation: String,
   ): Future[Unit] = {
     userManagementStore.getUser(userId).flatMap {
       case Right(user) if user.identityProviderId != identityProviderId =>
```

Comment:
Do you still need to check for idp now that this method is called `verifyUserDoesNotExist`?

Reply:
yes, we want to throw permission denied rather than user exists